### ISSUE TITLE
Support legacy templates for --generate-update-payload

### ISSUE THREAD

**Issue by chingis-elastic** [elastic/cloud-sdk-go#245]
## Readiness Checklist
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I am reporting the issue to the correct repository (for multi-repository projects)

## Expected Behavior
```
> ecl deployment show ${deployment_id} --generate-update-payload
```
should work on deployments with legacy templates

## Current Behavior
```
> ecl --config ecqa deployment show 8435137e1d624a1d9810728b8b382001 --generate-update-payload
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x18e76b3]
goroutine 1 [running]:
github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi.parseKibanaGetResponse(0xc0004cbc50, 0xc00060fab0, 0xd, 0x0)
    /Users/marc/go/pkg/mod/github.com/elastic/cloud-sdk-go@v1.0.1-0.20200903051513-9b118e6f0964/pkg/api/deploymentapi/update_payload.go:121 +0x123
github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi.NewUpdateRequest(0xc0004e4330, 0x7ffeefbff8b2)
    /Users/marc/go/pkg/mod/github.com/elastic/cloud-sdk-go@v1.0.1-0.20200903051513-9b118e6f0964/pkg/api/deploymentapi/update_payload.go:48 +0x31d
github.com/elastic/ecctl/cmd/deployment.glob..func8(0x3128f40, 0xc0001b4600, 0x1, 0x4, 0x0, 0x0)
    /Users/marc/go/pkg/mod/github.com/elastic/ecctl@v1.0.1-0.20200904065544-7e0a59b52f15/cmd/deployment/show.go:95 +0x527
github.com/spf13/cobra.(*Command).execute(0x3128f40, 0xc0001b45c0, 0x4, 0x4, 0x3128f40, 0xc0001b45c0)
    /Users/marc/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x3126540, 0x3107640, 0x2, 0x2)
    /Users/marc/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
    /Users/marc/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/elastic/cloud-cli/cmd.Execute(0x248667c, 0x5, 0x24ee440, 0x28, 0x248b610, 0x7, 0x248bf68, 0x9, 0x0)
    /Users/marc/repos/elastic/cloud-cli/cmd/root.go:79 +0x1cf
main.main()
    /Users/marc/repos/elastic/cloud-cli/main.go:35 +0x85
```

## Steps to Reproduce
1. create a deployment
2. edit Kibana configuration in advanced edit:
```
              "memory_per_node": 1024,
              "node_count_per_zone": 1,
              "instance_configuration_id": "aws.kibana.classic",
              "zone_count": 1,

```
3. run `ecl deployment show ${deployment_id} --generate-update-payload` and get above `parseKibanaGetResponse` stack

## Context
It prevented o11y team to use native `ecl` commands in upgrade script, so we resorted to using raw `ecl api 'v1/deployment'` calls 

## Your Environment
* Version used: 1.2.2 (Linux/MacOS)
* Link to your project: https://github.com/elastic/cloud/tree/master/found-deployment/docker/olly_upgrader

## Notes
Initially was reported in https://github.com/elastic/cloud/issues/69425 - it has more example failures (`parseElasticsearchGetResponse`, `parseApmGetResponse`)

**Comment by marclop** [comment]
I believe we don't plan on supporting legacy plans at all, since they're legacy. They should be converted to "normal" plans instead.

**Comment by marclop** [comment]
This has been actually fixed by #267
